src/platform/with_wgl/native_gl_context.rs

```diff
@@ -0,0 +1,144 @@
+use platform::NativeGLContextMethods;
+use std::ffi::CString;
+use std::os::raw::{c_void};
+use std::ptr;
+use std::sync::{Once, ONCE_INIT};
+
+use winapi;
+use user32;
+use kernel32;
+use super::wgl;
+use super::wgl_attributes::*;
+use gleam::gl;
+
+/// Wrapper to satisfy `Sync`.
+struct HMODULEWrapper(winapi::HMODULE);
+unsafe impl Sync for HMODULEWrapper {}
+
+lazy_static! {
+    static ref GL_LIB: Option<HMODULEWrapper>  = {
+        let p = unsafe{kernel32::LoadLibraryA(b"opengl32.dll\0".as_ptr() as *const _)};
+        if p.is_null() {
+            debug!("opengl32.dll not found!");
+            None
+        }
+        else {
+            debug!("opengl32.dll LOADED!");
+            Some(HMODULEWrapper(p))
+        }
+    };
+}
+
+static LOAD_GL: Once = ONCE_INIT;
+pub fn load_gl() {
```

Comment:
Another solution is to change the gl:: generator from global to struct based on this project. It requires a bit more work but it adds a lot of flexibility specially on windows were we can have multiple OpenGL implementations at the same time (for example angle and wgl). I want to add angle implementation for windows too and I had problems in servo because of the global gl. Struct based gl could be a good solution. What do you think?